fetools-app/src/components/ColorPicker/CustomPicker.jsx

```diff
@@ -3,72 +3,163 @@ import { getHexString } from "../ColorGradientComponents/ColorGradientUtils";
 
 export default function CustomPicker({
   colorData,
-  handleColorChange
+  handleColorChange,
+  handleQuery
 }){
 
+  const canvasContainerRef = useRef();
   const canvasRef = useRef();
   const markerRef = useRef();
   const intervalMouseMoveRef = useRef(null);
   const intervalMouseClickRef = useRef(null);
 
+  const [imgFile, setImgFile] = useState(null)
+  const [isColorPicker, setIsColorPicker] = useState(true)
   const [mouseCoor, setMouseCoor] = useState({x:0,y:0})
   const [currentColor, setCurrentColor] = useState(getHexString(colorData.color))
 
   useEffect(()=>{
     markerRef.current.children[0].style.background = colorData.color
-    createCanvasGradients()
     canvasRef.current.currentColor = currentColor
-
     return ()=>{
       stopInterval(intervalMouseMoveRef)
     }
-
   },[mouseCoor])
 
+  useEffect(()=>{
+    resizeCanvas()
+
+    if(isColorPicker){
+      createCanvasGradients()
+    } else{
+      console.log('img picker')
+      createImagePicker()
+    }
+
+  },[isColorPicker, imgFile])
+
   return(
     <>
     <div>
-      <div className="relative">
+      <div ref={canvasContainerRef} className="relative">
         <canvas 
         ref={canvasRef} 
         id="color-picker"
+        onDragOver={(e)=>isColorPicker?"":e.preventDefault()}
+        onDrop={handleOnDrop}
         onMouseMove={(e)=>startInterval(e,handleOnMouseMove,intervalMouseMoveRef)}
         onClick={(e)=>handleClick(e,true)}
         onMouseDown={(e)=>startInterval(e,handleClick,intervalMouseClickRef)}
-        onMouseUp={()=>(stopInterval(intervalMouseClickRef), stopInterval(intervalMouseMoveRef))}
+        onMouseUp={()=>(
+          stopInterval(intervalMouseClickRef), 
+          stopInterval(intervalMouseMoveRef), 
+          handleQuery(currentColor))}
         onMouseLeave={()=>(stopInterval(intervalMouseClickRef), stopInterval(intervalMouseMoveRef))}
```

Comment:
We need to add logic so that if the mouse leaves the canvas but then re-enters without a mouseUp event, the picker continues to function.